### PR TITLE
Feat/move reducers to TS

### PR DESCRIPTION
**Pull request by alvaro-ricotta**
Reducers have been updated to TypeScript.
We implemented simple typings to better understand what type of data we expect in each reducer. Additionally, we changed the previous reducer's switch case-based approach to a config-driven version.

Finally, we modified the methods to select the handler based on the previously established config-driven approach.

### CODE REVIEW

File: packages/boba/gateway/src/reducers/loginReducer.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program. If not, see <https://www.gnu.org/licenses/>.
```
Comment (InoMurko) [inline packages/boba/gateway/src/reducers/loginReducer.ts]: eh?? :D

File: packages/boba/gateway/src/reducers/notificationReducer.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
```
Comment (InoMurko) [inline packages/boba/gateway/src/reducers/notificationReducer.ts]:
```suggestion

```

**Comment by alvaro-ricotta** [comment]
All disclaimer has been updated @InoMurko

File: packages/boba/gateway/src/reducers/loginReducer.ts
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/loginReducer.ts:1]: We are not using the LoginReducer in application, please remove this!

File: packages/boba/gateway/src/reducers/notificationReducer.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+const notificationReducer = (
+  state: State = initialState,
+  action: Action
+): State => {
+  const handler = actionHandlers[action.type]
+  return handler ? handler(state, action) : state
+}
+
+export default notificationReducer
```
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/notificationReducer.ts:68]: This block code seems repeated, so can we move it to separate file and reuse where it's needed!

File: packages/boba/gateway/src/reducers/signatureReducer.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+type State = {
+  exitLPsigned: boolean
+  exitTRADsigned: boolean
+  depositLPsigned: boolean
+}
+
+type ActionTypes = 'EXIT/LP/SIGNED' | 'EXIT/TRAD/SIGNED' | 'DEPOSIT/LP/SIGNED'
```
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/signatureReducer.ts:22]: Please move the constants to separate file or variable at top.
like 
```
const EXIT_LP_SIGNED = 'EXIT/LP/SIGNED'

/// refer the same for defining the types and actions types in reducere also let's use the same in selectors too.
```

File: packages/boba/gateway/src/reducers/transactionReducer.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  | 'TRANSFER/CREATE/SUCCESS'
+  | 'SETTLE_v0/CREATE/SUCCESS'
+  | 'SETTLE_v1/CREATE/SUCCESS'
+  | 'SETTLE_v2/CREATE/SUCCESS'
+  | 'SETTLE_v2OLO/CREATE/SUCCESS'
+  | 'SETTLE_v3/CREATE/SUCCESS'
+  | 'SETTLE_v3OLO/CREATE/SUCCESS'
+  | 'MONSTER/CREATE/SUCCESS'
+  | 'CDM/COMPLETE/SET'
+  | 'CDM/COMPLETE/RESET'
```
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/transactionReducer.ts:33]: Like mention above, define constant var and use it across app.

File: packages/boba/gateway/src/reducers/transactionReducer.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+const initialState: State = {
+  CDMType: '',
+  CDMMessage: '',
+  CDMTransaction: '',
+}
+
+const actionHandlers: Record<string, (state: State, action: Action) => State> =
+  {
```
Comment (sk-enya) [inline packages/boba/gateway/src/reducers/transactionReducer.ts:47]: Listener for the below actions in reducer is missing.

```
SETTLE_v0/CREATE/SUCCESS'
  | 'SETTLE_v1/CREATE/SUCCESS'
  | 'SETTLE_v2/CREATE/SUCCESS'
  | 'SETTLE_v2OLO/CREATE/SUCCESS'
  | 'SETTLE_v3/CREATE/SUCCESS'
  | 'SETTLE_v3OLO/CREATE/SUCCESS'

```

**Review by sk-enya — CHANGES_REQUESTED** [review]
As this pr is completely for reducer can you please migrate the `reducers/index.js` file to typescript with required types?

**Comment by InoMurko** [comment]
Reopen if fixed.

**Pull request closed without merging**